src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php

```diff
@@ -309,6 +309,16 @@ public function from(string $url)
         return $this->withHeader('referer', $url);
     }
 
+    /**
+     * Set the precognition header
+     *
+     * @return $this
+     */
+    public function usingPrecognition()
+    {
+        return $this->withHeader('Precognition', 'true');
+    }
+
```

Comment:
Casual suggestion: For testing helpers like this we generally use the `withX`. So we could use:

```suggestion
    /**
     * Set the precognition header
     *
     * @return $this
     */
    public function withPrecognition()
    {
        return $this->withHeader('Precognition', 'true');
    }

```